### PR TITLE
Feature/angular 11

### PR DESCRIPTION
**Pull request by laneli**
Updates Angular to 11.2.8 (because of peer dependency requirements for barista) and barista to 9.6.0

### CODE REVIEW

File: bridge/client/app/app.routing.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ const routes: Routes = [
 ];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [RouterModule.forRoot(routes, { relativeLinkResolution: 'legacy' })],
```
Comment (christian-kreuzberger-dtx) [inline bridge/client/app/app.routing.ts:35]: please add a docstring telling what `relativeLinkResolution` is doing

File: bridge/client/main.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import 'hammerjs';
```
Comment (christian-kreuzberger-dtx) [inline bridge/client/main.ts:1]: Is this related to the Angular Upgrade, or did we actually not need this at all?

File: bridge/e2e/tsconfig.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "../out-tsc/e2e",
     "module": "commonjs",
-    "target": "es5",
+    "target": "es2018",
```
Comment (christian-kreuzberger-dtx) [inline bridge/e2e/tsconfig.json:6]: Is this a requirement? AFAIK `es2018` is still not supported by all browsers.

File: bridge/karma-headless.conf.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ module.exports = function (config) {
       require('karma-jasmine'),
       require('karma-chrome-launcher'),
       require('karma-jasmine-html-reporter'),
-      require('karma-coverage-istanbul-reporter'),
+      require('karma-coverage'),
```
Comment (christian-kreuzberger-dtx) [inline bridge/karma-headless.conf.js:13]: Why changing the reporter?

File: bridge/karma.conf.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ module.exports = function (config) {
       require('karma-jasmine'),
       require('karma-chrome-launcher'),
       require('karma-jasmine-html-reporter'),
-      require('karma-coverage-istanbul-reporter'),
+      require('karma-coverage'),
```
Comment (christian-kreuzberger-dtx) [inline bridge/karma.conf.js:15]: Same here, why changing the reporter?

File: bridge/karma-headless.conf.js (hunk 4 above)
Comment (laneli) [reply bridge/karma-headless.conf.js:13]: Because it is deprecated

File: bridge/karma.conf.js (hunk 5 above)
Comment (laneli) [reply bridge/karma.conf.js:15]: Same here, deprecated

File: bridge/tsconfig.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
     "declaration": false,
     "downlevelIteration": true,
     "experimentalDecorators": true,
-    "module": "esnext",
+    "module": "es2020",
```
Comment (christian-kreuzberger-dtx) [inline bridge/tsconfig.json:10]: Not sure what this does, can you please explain?

File: bridge/client/main.ts (hunk 2 above)
Comment (laneli) [reply bridge/client/main.ts:1]: https://indepth.dev/posts/1075/exciting-times-ahead-be-ready-for-angular-9#hammerjs-is-now-optional
It was removed by the update process. Also hammerjs is not used in bridge.

File: bridge/client/app/app.routing.ts (hunk 1 above)
Comment (laneli) [reply bridge/client/app/app.routing.ts:35]: Migration added this automatically.
Angular 11 introduced a bugfix for the usage of relative paths. https://angular.io/api/router/ExtraOptions#relativeLinkResolution
We are not using that anywhere - so we could get rid of it.

File: bridge/e2e/tsconfig.json (hunk 3 above)
Comment (laneli) [reply bridge/e2e/tsconfig.json:6]: I think this is not relevant for browsers, as this is located in the e2e folder and used for e2e tests.

File: bridge/tsconfig.json (hunk 6 above)
Comment (laneli) [reply bridge/tsconfig.json:10]: https://angular.io/guide/migration-update-module-and-target-compiler-options

**Review by christian-kreuzberger-dtx — DISMISSED** [review]
thanks for the answers, approved!

**Comment by laneli** [comment]
Also please note that two npm audit vulnerabilities are left in on purpose. At the moment it is not feasible to update highcharts. Also postcss can not be added to resolutions with a fixed version without breaking tests.

File: bridge/client/app/app.routing.ts (hunk 1 above)
Comment (laneli) [reply bridge/client/app/app.routing.ts:35]: This will be removed, as it is not needed in Bridge

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**